### DISCUSSION TITLE
Why is some table not showing primary keys ?

### DISCUSSION THREAD

**Discussion by dittorahmat** [nocodb/nocodb#929] [Q&A] [answered]
I have some tables from MySQL (data is generated from app or scraped from web). All of them have primary keys

But somehow nocodb only recognize primary keys on some of the table. Some of them aren't recognized, which disabled updating cell. When I click on a cell to try to update, error is shown : **Update not allowed for table which doesn't have primary key** . Which is not true, because I already set primary key in MySQL.

Is there any solution for this ?

**Comment by o1lab** [comment, marked as answer]
Are you sure of a primary key exists for those tables ? Or were these primary keys added after you created the project ?

If you open fields (right top) - you can enable system fields and see what are the primary keys in grid view .

**Reply by dittorahmat** [reply]
I think some of the primary key was created after I created the project. Was this the case ? If so, how do I get nocodb to recognize these primary keys ? Do I need to recreate the project again from scratch ? 

I assumed tables refresh (right click on tables) would be enough. Since all of new tables or deleted tables will be updated when I click this

**Reply by o1lab** [reply]
Schema changes needs to be synced : https://docs.nocodb.com/setup-and-usages/sync-schema
